plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPage.java

```diff
@@ -1088,8 +1101,27 @@ public boolean isPageComplete() {
 
 	public IServer saveServer(IProgressMonitor monitor) throws CoreException {
 		model.updateServer();
+		setOpenshiftMavenProfileActive(model.getDeployProject());
 		return model.saveServer(monitor);
 	}
+	
+	@SuppressWarnings("restriction")
+    private void setOpenshiftMavenProfileActive(IProject newDeployProject)  throws CoreException  {
+	    IFile pom = newDeployProject.getFile(IMavenConstants.POM_FILE_NAME);
+	    if (newDeployProject.hasNature(IMavenConstants.NATURE_ID) && pom != null) {                   
+            final IProfileManager profileManager = MavenProfilesCoreActivator.getDefault().getProfileManager();
+            IMavenProjectFacade facade = MavenPlugin.getMavenProjectRegistry().create(pom, true, new NullProgressMonitor());
+            List<ProfileData> profiles = profileManager.getProfileDatas(facade, new NullProgressMonitor());
+            List<String> activeProfiles = profiles.stream()
+                    .filter(p -> ProfileState.Active.equals(p.getActivationState()))
```

Comment:
writing down what we discussed/found out:
with a ~/.m2/settings.xml file, that also has profiles, filtering only on activation states this will have all those activated:
![image](https://user-images.githubusercontent.com/25126/43846791-7b7d852c-9b2f-11e8-98c5-7052ab2be934.png)
One needs to also filter autoactive profiles:
```
.filter(p -> ProfileState.Active.equals(p.getActivationState())
                        && !p.isAutoActive())
```
This will produce the correct results:
![image](https://user-images.githubusercontent.com/25126/43846958-d3619efe-9b2f-11e8-8c9c-d22994ce85e4.png)
![image](https://user-images.githubusercontent.com/25126/43847072-10bb95b6-9b30-11e8-968c-29c7f9d3377c.png)